src/UglyToad.PdfPig/Writer/PdfMerger.cs

```diff
@@ -122,369 +110,39 @@ public static void Merge(IReadOnlyList<Stream> streams, Stream output, IReadOnly
-                        return;
+                        pages = pagesBundle[fileIndex];
                     }
 
-                    pageIndices = Enumerable.Range(1, pagesCount);
-                }
-                else if (pages.Count < 1)
-                {
-                    return;
-                }
-                else
-                {
-                    pageIndices = pages;
-                }
-
-                currentVersion = Math.Max(version, currentVersion);
-
-                var referencesFromDocument = new Dictionary<IndirectReference, IndirectReferenceToken>();
-
-                var currentNodeReference = context.ReserveNumberToken();
-                var pagesReferences = new List<IndirectReferenceToken>();
-                var resources = new Dictionary<string, IToken>();
-
-                bool DoesAEntryCollide(PageTreeNode node)
-                {
-                    while (node != null)
+                    if (pages == null)
                     {
-                        var dictionary = node.NodeDictionary;
-                        if (dictionary.TryGet(NameToken.Resources, tokenScanner, out DictionaryToken resourcesDictionary))
+                        for (var i = 1; i <= existing.NumberOfPages; i++)
                         {
-                            var nonCollidingResources = resourcesDictionary.Data.Keys.Except(resources.Keys);
-                            if (nonCollidingResources.Count() != resourcesDictionary.Data.Count)
-                            {
-                                // This means that at least one of the resources collided
-                                return true;
-                            }
+                            document.AddPage(existing, 1);
                         }
-
-                        /* TODO: How to handle?
-                         *  `Rotate`
-                         *  `CropBox`
-                         *  `MediaBox`
-                         */
-
-                        // No colliding entry was found, in this node
-                        // Keep walking up into the tree
-                        node = node.Parent;
-                    }
-
-                    return false;
-                }
-
-
-                void CopyEntries(PageTreeNode node)
-                {
-                    while (node != null)
+                    } else
                     {
-                        var dictionary = node.NodeDictionary;
-                        if (dictionary.TryGet(NameToken.Resources, tokenScanner, out DictionaryToken resourcesDictionary))
+                        foreach (var i in pages)
                         {
-                            foreach (var pair in resourcesDictionary.Data)
-                            {
-                                resources.Add(pair.Key, CopyToken(pair.Value, tokenScanner, referencesFromDocument));
-                            }
+                            document.AddPage(existing, 1);
```

Comment:
this should be i

Reply:
good catch, adding test case for merger that will cover.